File: Atividades A1/15 Questões/Python/Questoes15/Questao10/Main.py
#10. Sobrecarga de Métodos (Java) / Métodos com Nomes Diferentes (Python, Go) Implemente uma classe Calculadora com métodos somar que aceita diferentes números de parâmetros (dois ou três números).
#Em Golang, use funções com nomes diferentes para diferentes quantidades de parâmetros.

class Calculadora:
    # Método para somar dois números
    def somarDois(self, a, b):
        return a + b

    # Método para somar três números
    def somarTres(self, a, b, c):
        return a + b + c

if __name__ == "__main__":
    calculadora = Calculadora()

    # Chamadas aos métodos da calculadora
    x1 = calculadora.somarDois(14, 11)
    print("14 + 11 =", x1)

    x2 = calculadora.somarTres(10, 5, 90)
    print("10 + 5 + 90 =", x2)

File: Atividades A1/15 Questões/Java/Questoes15/src/Questao10/Main.java
package Questao10;
//10. Sobrecarga de Métodos (Java) / Métodos com Nomes Diferentes (Python, Go) Implemente uma classe Calculadora com métodos somar que aceita diferentes números de parâmetros (dois ou três números).
//Em Golang, use funções com nomes diferentes para diferentes quantidades de parâmetros.

class Calculadora {

    // Método para somar dois números
    public int somar(int a, int b) {
        return a + b;
    }

    // Método para somar três números
    public int somar(int a, int b, int c) {
        return a + b + c;
    }
}
public class Main {
    public static void main(String[] args) {
        Calculadora calculadora = new Calculadora();

        // Chamadas aos métodos da calculadora
        int x1 = calculadora.somar(14, 11);
        System.out.println("14 + 11 = " + x1);

        int x2 = calculadora.somar(10, 5, 90);
        System.out.println("10 + 5 + 90 = " + x2);
    }
}